src/managers/PermissionOverwriteManager.js

```diff
@@ -0,0 +1,138 @@
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * // Edit or Create permission overwrites for a message author
+   * message.channel.permissionOverwrites.edit(message.author, {
+   *   SEND_MESSAGES: false
+   * })
+   *   .then(channel => console.log(channel.permissionOverwrites.cache.get(message.author.id)))
+   *   .catch(console.error);
+   */
+  async edit(userOrRole, options, reason) {
+    userOrRole = this.channel.guild.roles.resolve(userOrRole) ?? this.client.users.resolve(userOrRole);
+    if (!userOrRole) throw new TypeError('INVALID_TYPE', 'parameter', 'User nor a Role');
+
+    const type = userOrRole instanceof Role ? OverwriteTypes.role : OverwriteTypes.member;
+    const { allow, deny } = PermissionOverwrites.resolveOverwriteOptions(options);
+
+    const existing = this.cache.get(userOrRole.id);
+    if (existing) {
+      await this.client.api
+        .channels(this.channel.id)
+        .permissions(userOrRole.id)
+        .put({
+          data: { id: userOrRole.id, type, allow, deny },
+          reason,
+        });
+      return this.channel;
+    }
+    return this.create(userOrRole, options, reason);
+  }
+
+  /**
+   * Creates permission overwrites for a user or role in this channel, or replaces them if already present.
+   * @param {RoleResolvable|UserResolvable} userOrRole The user or role to update
+   * @param {PermissionOverwriteOptions} options The options for the update
+   * @param {string} [reason] Reason for creating/editing this overwrite
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * // Create or Replace permission overwrites for a message author
+   * message.channel.permissionOverwrites.create(message.author, {
+   *   SEND_MESSAGES: false
+   * })
+   *   .then(channel => console.log(channel.permissionOverwrites.cache.get(message.author.id)))
+   *   .catch(console.error);
+   */
+  async create(userOrRole, options, reason) {
+    userOrRole = this.channel.guild.roles.resolve(userOrRole) ?? this.client.users.resolve(userOrRole);
+    if (!userOrRole) throw new TypeError('INVALID_TYPE', 'parameter', 'User nor a Role');
+
+    const type = userOrRole instanceof Role ? OverwriteTypes.role : OverwriteTypes.member;
+    const { allow, deny } = PermissionOverwrites.resolveOverwriteOptions(options);
+
+    await this.client.api
+      .channels(this.channel.id)
+      .permissions(userOrRole.id)
+      .put({
+        data: { id: userOrRole.id, type, allow, deny },
+        reason,
+      });
+    return this.channel;
+  }
+
+  /**
+   * Deletes permission overwrites for a user or role in this channel
+   * @param {UserResolvable|RoleResolvable} userOrRole The user or role to delete
+   * @param {string} [reason] The reason for deleting the overwrite
+   * @returns {GuildChannel}
+   */
+  async delete(userOrRole, reason) {
+    userOrRole = this.channel.guild.roles.resolve(userOrRole) ?? this.client.users.resolve(userOrRole);
```

Comment:
Should use IDs, not full objects. That way, for example, we can pass in an ID from a db without needing to first fetch the user